front/src/applications/operationalStudies/components/MacroEditor/nodeStore.ts

```diff
@@ -0,0 +1,49 @@
+type Node = {
+  trigram: string;
+  positionX: number;
+  positionY: number;
+};
+
+const NODES_LOCAL_STORAGE_KEY = 'macro_nodes';
+
+let nodesCache: Map<string, Node> | null;
+
+const loadNodes = () => {
+  if (nodesCache) return nodesCache;
+
+  let entries = [];
+  const rawNodes = localStorage.getItem(NODES_LOCAL_STORAGE_KEY);
+  if (rawNodes) {
+    entries = JSON.parse(rawNodes).map((node: Node) => [node.trigram, node]);
+  }
+
+  nodesCache = new Map(entries);
+  return nodesCache;
+};
+
+const saveNodes = (nodes: Map<string, Node>) => {
+  localStorage.setItem(NODES_LOCAL_STORAGE_KEY, JSON.stringify([...nodes.values()]));
+};
```

Comment:
When the page is refreshed, our saved nodes no longer have their saved positions.

```suggestion
const loadNodes = () => {
  if (nodesCache) return nodesCache;

  let entries = [];
  const rawNodes = localStorage.getItem(NODES_LOCAL_STORAGE_KEY);
  if (rawNodes) {
    entries = JSON.parse(rawNodes).map((node: Node) =>  node);
  }

  nodesCache = new Map(entries);
  return nodesCache;
};

const saveNodes = (nodes: Map<string, Node>) => {
  localStorage.setItem(NODES_LOCAL_STORAGE_KEY, JSON.stringify([...nodes]));
};
```

Reply:
done thanks